tests/Microsoft.Identity.Test.Unit.net45/TestExtensions.cs

```diff
@@ -27,13 +30,35 @@ public static class TestExtensions
             Assert.AreEqual(expectedAppMetadataCount, accessor.GetAllAppMetadata().Count());
         }
 
-        public static void InitializeTokenCacheFromFile(this IPublicClientApplication pca, string resourceFile)
+        public static void InitializeTokenCacheFromFile(this IPublicClientApplication pca, string resourceFile, bool updateATExpiry = false)
         {
             string tokenCacheAsString = File.ReadAllText(resourceFile);
+
+            if (updateATExpiry)
+            {
+                var cacheJson = JObject.Parse(tokenCacheAsString);
+
+                
+
+                JEnumerable<JToken> tokens = cacheJson["AccessToken"].Children();
+                foreach (JToken token in tokens)
+                {
+                    var obj = token.Children().Single() as JObject;
+                    //obj["cached_at"] = DateTime.UtcNow + TimeSpan.FromMinutes(100);
+                    //CoreHelpers.DateTimeToUnixTimestamp(DateTimeOffset.Now.AddMinutes(100))
+                    obj["expires_on"] = CoreHelpers.DateTimeToUnixTimestamp(DateTimeOffset.Now.AddMinutes(100));
```

Comment:
Fancy new way of populating the token cache from a file :) - this makes the test much more readable than mocking dozens of http calls to populate the cache.